minijinja/src/value/object.rs

```diff
@@ -101,3 +88,230 @@ impl<T: Object> Object for std::sync::Arc<T> {
         T::call(self, state, args)
     }
 }
+
+/// A kind defines the object's behavior.
+///
+/// When a dynamic [`Object`] is implemented, it can be of one of the kinds
+/// here.  The default behavior will be a [`Basic`](Self::Basic) object which
+/// doesn't do much other than that it can be printed.  For an object to turn
+/// into a [struct](Self::Struct) or [sequence](Self::Seq) the necessary kind
+/// has to be returned with a pointer to itself.
+///
+/// Today object's can have the behavior of structs and sequences but this
+/// might expand in the future.  It does mean that not all types of values can
+/// be represented by objects.
+#[non_exhaustive]
+pub enum ObjectKind<'a> {
+    /// This object is a basic object.
+    ///
+    /// Such an object has no attributes but it might be callable and it
+    /// can be stringified.  When serialized it's serialized in it's
+    /// stringified form.
+    Basic,
```

Comment:
`Basic` doesn't provide a lot of meaning for me. Is this name coming from another ecosystem? Given the description, I would expect this to be called something like `Function` or `Callable`.

Reply:
Maybe `Plain`? A basic object is not necessarily callable. The most basic behavior it has is just that if you print it with `{{ object }}` it is rendered as string.